### PR TITLE
(storage): impl some function

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
added some implementation for todo! function in mock.rs in storage

### CODE REVIEW

File: crates/storage/db/src/abstraction/mock.rs [hunk 1]
```diff
@@ -105,31 +105,42 @@ pub struct CursorMock {
 
 impl<T: Table> DbCursorRO<T> for CursorMock {
     fn first(&mut self) -> PairResult<T> {
-        todo!()
+        self._cursor = 0;
+        Ok(None)
```
Comment (mattsse) [inline crates/storage/db/src/abstraction/mock.rs:109]: I'm okay with returning None everywhere,

but let's remove the cursor updates, don't think we need them

File: crates/storage/db/src/abstraction/mock.rs [hunk 2]
```diff
@@ -105,31 +105,42 @@ pub struct CursorMock {
...
 
     fn seek_exact(&mut self, _key: T::Key) -> PairResult<T> {
-        todo!()
+        let mut rng = rand::thread_rng();
+        let found = rng.gen_bool(0.5);
+        if found {
+            Err(DatabaseError::Read(-1))
+        } else {
+            Ok(None)
+        }
```
Comment (mattsse) [inline crates/storage/db/src/abstraction/mock.rs:119]: just returning OkNone is fine

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/storage/db/src/abstraction/mock.rs (hunk 1 above)
Comment (DoTheBestToGetTheBest) [reply crates/storage/db/src/abstraction/mock.rs]: > I'm okay with returning None everywhere,
> 
> but let's remove the cursor updates, don't think we need them

ty lot Mat!

File: crates/storage/db/src/abstraction/mock.rs (hunk 2 above)
Comment (DoTheBestToGetTheBest) [reply crates/storage/db/src/abstraction/mock.rs]: > just returning OkNone is fine

ty lot!

**Review by mattsse — APPROVED** [review]

**Pull request merged**